main/src/hadd.cxx

```diff
@@ -378,48 +615,59 @@ int main( int argc, char **argv )
    }
    // The following section will collect all input filenames into a vector,
    // including those listed within an indirect file.
-   // If any file can not be accessed, it will error out, unless skip_errors is true
+   // If any file can not be accessed, it will error out, unless args.fSkipErrors is true
    std::vector<std::string> allSubfiles;
-   for (int a = ffirst; a < argc; ++a) {
-      if (a == outputPlace)
-         continue;
+   for (int a = args.fFirstInputIdx; a < argc; ++a) {
+      if (!args.fNoFlagsAfterPositionalArguments && argv[a] && argv[a][0] == '-') {
+         break;
+      }
       if (argv[a] && argv[a][0] == '@') {
          std::ifstream indirect_file(argv[a] + 1);
          if (!indirect_file.is_open()) {
-            std::cerr << "hadd could not open indirect file " << (argv[a] + 1) << std::endl;
-            if (!skip_errors)
+            Err() << "could not open indirect file " << (argv[a] + 1) << std::endl;
+            if (!args.fSkipErrors)
                return 1;
          } else {
             std::string line;
             while (indirect_file) {
-               if( std::getline(indirect_file, line) && line.length() ) {
+               if (std::getline(indirect_file, line) && line.length()) {
                   if (gSystem->AccessPathName(line.c_str(), kReadPermission) == kTRUE) {
-                     std::cerr << "hadd could not validate the file name \"" << line << "\" within indirect file "
+                     Err() << "could not validate the file name \"" << line << "\" within indirect file "
                                << (argv[a] + 1) << std::endl;
-                     if (!skip_errors)
+                     if (!args.fSkipErrors)
+                        return 1;
+                  } else if (line == targetname) {
+                     Err() << "file " << line << " cannot be both the target and an input!\n";
+                     if (!args.fSkipErrors)
                         return 1;
-                  } else
+                  } else {
                      allSubfiles.emplace_back(line);
+                  }
                }
             }
          }
       } else {
          const std::string line = argv[a];
          if (gSystem->AccessPathName(line.c_str(), kReadPermission) == kTRUE) {
-            std::cerr << "hadd could not validate argument \"" << line << "\" as input file " << std::endl;
-            if (!skip_errors)
+            Err() << "could not validate argument \"" << line << "\" as input file " << std::endl;
+            if (!args.fSkipErrors)
                return 1;
-         } else
+         } else if (line == targetname) {
```

Comment:
Probably overkill :)
```suggestion
         } else if (std::filesystem::equivalent(line, targetname)) {
```
(note: `equivalent` throws if any of the files do not exist, however here we already know they exist).